### PR TITLE
Add any-db in the backend (closes #19)

### LINKED ISSUE #19: Data store
We need to add a database to the mix, I was thinking on PostgreSQL (as it is already in heroku) but it could be great to make it work with SQLite in development (as it is better for development).

May be there are better options to keep the data, but this sounds pretty standard.

- [x] 1 -Create a new branch named *19-anydb*.
- [x] 2 - Assemble a script that creates a sqlite database and a seniorities table with rows *id*, *name* and *skills*.
- [x] 3 -  Assemble a script that loads ```./seniorities.json``` inside the newly created sqlite seniorities table.
- [x] 4 - Modify lines https://github.com/Altoros/hacker-assessor/blob/master/server.js#L19 and similar to call a findSeniority(‘sth’) function preserving current functionality. The function findSeniority should be something like ```require(‘**seniorities.js’).findSeniority```. This should be the last change in the ```./server.js``` of this branch.
- [x] 5 - Modify the function findSeniority from step 4 to make it search inside the sqlite database.
- [x] 6 - Make sure things work the same way it worked before.

### PR DESCRIPTION
**Pull request by eloyesp**
Move seniorities to a SQL database (sqlite3) using any-db.

### CODE REVIEW

File: migrations/create-seniorities-table.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var seniorities = require("../models/seniorities");
+var anyDB = require('any-db');
+var conn = anyDB.createConnection('sqlite3://hackerassessor.sqlite3');
+var query = seniorities.create().toQuery();
```
Comment (eloyesp) [inline migrations/create-seniorities-table.js]: Could you extract the connection out?

Comment (n370) [inline migrations/create-seniorities-table.js]: Yep, any idea? I'm calling you.

File: migrations/load-seniorities.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  query = seniorities.insert({
+    name: item.name,
+    skills: item.skills.join(",")
+  }).toQuery();
+
+  conn.query(query.text, query.values, function(error, result) {
+    if (error) {
+      console.dir(error);
+    } else {
+      console.dir(result);
```
Comment (eloyesp) [inline migrations/load-seniorities.js]:
```javascript
console.dir(error || result);
// or
conn.query(query.text, query.values, console.log);
```

Comment (n370) [inline migrations/load-seniorities.js]: Sounds good!

File: database.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Create a database connection.
+ * @param {string} [environment] An environment name. If none is passed, defaults to the
+ *                               development environment.
+ */
+function connect(environment) {
+  switch (environment) {
+    default:
+      return anyDB.createConnection('sqlite3://hackerassessor.sqlite3');
```
Comment (n370) [inline database.js]: @eloyesp I'm wondering if we could somehow cache the connection created here. Is that something we should even consider at this stage of development?

**Comment by n370** [comment]
@eloyesp please review my last commits then, if everything looks fine please send me a green light to merge. Thanks a lot.

**Pull request merged**